packages/@sfx/core/src/utils/dependencies.ts

```diff
@@ -0,0 +1,93 @@
+import { Plugin } from '../plugin';
+
+/**
+ * The type of the plugin dependency graph. The dependency graph is a
+ * directed graph whose vertices are plugins and whose edges are
+ * "depended-on" relations. For example, an edge going from plugin A to
+ * plugin B means that A is depended on by B (that is, B depends on A).
+ *
+ * The graph is represented as an adjacency list, which itself is
+ * represented as an object whose keys are plugin names and whose values
+ * are lists of the names of dependent plugins.
+ */
+export interface DependencyGraph {
+  /** The names of each plugin's dependers. */
+  [name: string]: string[];
+}
+
+/**
+ * Creates an empty dependency graph.
+ *
+ * @returns An empty dependency graph.
+ * @hidden
+ */
+function createEmptyGraph(): DependencyGraph {
+  return Object.create(null);
```

Comment:
Any reason why this shouldn't be an empty array? I would consider that the 'trivial' dependency graph - there are no plugins or dependencies.

Reply:
Yeah, I realized it on second review today. Thanks!